packages/react-instantsearch/src/core/createInstantSearchManager.test.js

```diff
@@ -121,26 +121,32 @@ describe('createInstantSearchManager', () => {
   });
 
   describe('getWidgetsIds', () => {
-    it('returns the list of ids of all registered widgets', () => {
+    it('returns the list of ids of all registered widgets', done => {
       const ism = createInstantSearchManager({
         indexName: 'index',
         initialState: {},
         searchParameters: {},
         algoliaClient: client,
       });
 
+      const widgetIDsT0 = ism.getWidgetsIds().sort();
+      expect(widgetIDsT0).toEqual([]);
+
       ism.widgetsManager.registerWidget({getMetadata: () => ({id: 'a'})});
       ism.widgetsManager.registerWidget({getMetadata: () => ({id: 'b'})});
       ism.widgetsManager.registerWidget({getMetadata: () => ({id: 'c'})});
-      ism.widgetsManager.registerWidget({getMetadata: () => ({id: 'd'})});
 
-      const widgetIDsT0 = ism.getWidgetsIds().sort();
-      expect(widgetIDsT0).toEqual([]);
-
-      jest.runAllTimers();
+      ism.widgetsManager.registerWidget({getMetadata: () => ({id: 'd'})});
 
-      const widgetIDsT1 = ism.getWidgetsIds().sort();
-      expect(widgetIDsT1).toEqual(['a', 'b', 'c', 'd']);
+      Promise.resolve().then(() => {
+        try {
+          const widgetIDsT1 = ism.getWidgetsIds().sort();
```

Comment:
Did you try to write a simple test case without any of our code just to try return Promise.resolve().then(throw..)? maybe there's a bug somewhere or a misunderstanding of ours on how promise works with jest